product_variant_configurator/models/product_product.py

```diff
@@ -83,20 +83,33 @@ def _check_configuration_validity(self):
         """The method checks that the current selection values are correct.
 
         As default, the validity means that all the attributes
-        values are set. This can be overridden to set another rules.
+        with the required flag are set.
+
+        This can be overridden to set another rules.
 
         :raises: exceptions.ValidationError: If the check is not valid.
         """
         # Creating from template variants attributes are not created at once so
         # we avoid to check the constrain here.
         if self.env.context.get('creating_variants'):
             return
+        attribute_line_model = self.env['product.attribute.line']
         for product in self:
-            if bool(product.product_tmpl_id.attribute_line_ids.mapped(
-                    'attribute_id') -
-                    product.attribute_value_ids.mapped('attribute_id')):
+            errors = []
+            attribute_lines = product.product_tmpl_id.attribute_line_ids
+            for value in product.attribute_value_ids:
+                line = attribute_line_model.search([
```

Comment:
@pedrobaeza Your implementation is not scalable and will introduce performance issue on products import.
You should avoid to call `search` into a double loop!

Reply:
@pedrobaeza 
Why not 
```python
if product.product_tmpl_id.attribute_line_ids.mapped('attribute_id').filtered(lambda a: a.required) -  product.attribute_value_ids.mapped('attribute_id')):
    raise ...
```

Reply:
Yeah, it works with a slight variation (first `filtered` and then `mapped`). Thanks for the suggestion.